Jenkinsfile

```
@@ -6,9 +6,19 @@ for (int i = 0; i < (BUILD_NUMBER as int); i++) {
 }
 
 def branches = [:]
-
 def splits
-if (env.BUILD_NUMBER == '1') {
+def needSplittingFromWorkspace = true
+
+for (build = currentBuild.previousCompletedBuild; build != null; build = build.previousCompletedBuild) {
+    if (build.resultIsBetterOrEqualTo("UNSTABLE")) {
+        // we have a reference build
+        echo "not splitting from workkspace, reference build should be : ${build.previousNotFailedBuild.projectName}:${build.previousNotFailedBuild.number}, with state ${build.previousNotFailedBuild.result}"
```

When I tried to run this fragment in a pipeline job that I created, it failed with an error that the build.previousNotFailedBuild was null.  Does the echo need some form of safeguards to handle cases where there is no previous not failed build?
```suggestion
        echo "not splitting from workkspace, reference build should be : ${build.previousNotFailedBuild.projectName}:${build.previousNotFailedBuild.number}, with state ${build.previousNotFailedBuild.result}"
```

---

urgh... nope...

```suggestion
        echo "not splitting from workkspace, reference build should be : ${build.projectName}:${build.number}, with state ${build.result}"
```